### PR TITLE
[14.0][MIG] sentry: migrate sentry-raven to new api sentry-sdk

### PR DESCRIPTION
**Pull request by paradoxxxzero**
This is a migration done by cherry-picking https://github.com/OCA/server-tools/pull/1942 on top of current 14.0.
This superseeds #2223

### CODE REVIEW

File: sentry/__manifest__.py [hunk 1]
```diff
@@ -3,16 +3,25 @@
 {
     "name": "Sentry",
     "summary": "Report Odoo errors to Sentry",
-    "version": "14.0.1.0.2",
+    "version": "14.0.1.0.0",
```
Comment (legalsylvain) [inline sentry/__manifest__.py]: that version is incorrect.

File: sentry/__manifest__.py [hunk 2]
```diff
@@ -3,16 +3,25 @@
...
     "category": "Extra Tools",
     "website": "https://github.com/OCA/server-tools",
     "author": "Mohammed Barsi,"
     "Versada,"
     "Nicolas JEUDY,"
-    "Odoo Community Association (OCA)",
+    "Odoo Community Association (OCA),"
+    "Vauxoo",
+    "maintainers": ["barsi", "naglis", "versada", "moylop260", "fernandahf"],
```
Comment (legalsylvain) [inline sentry/__manifest__.py:14]: I think there are no rules regarding maintainers quantity, but maybe 5 maintainers is a little too much ? 

CC : @OCA/community-maintainers

**Review by legalsylvain — COMMENTED** [review]
partial review. 

thanks for porting this must-have update.

File: sentry/__manifest__.py (hunk 1 above)
Comment (paradoxxxzero) [reply sentry/__manifest__.py]: Good catch

**Comment by paradoxxxzero** [comment]
Thanks for the partial review.

I added some more fixes as explained in commits, most notably I added one more wsgi patch because as it is the `SentryWsgiMiddleware` was simply ignored by odoo. This hopefully will fix the performance monitoring.
Of course I'll rebase after review if it is deemed necessary.

**Comment by andreschenkels** [comment]
I reviewed the module and it seems to be working fine. Nice information and no longer depending on the raven module!
Please merge

**Review by hparfr — APPROVED** [review]
I'm using it in production since weeks

LGTM

**Comment by hparfr** [comment]
Can you please rebase ?

**Comment by andreschenkels** [comment]
@legalsylvain  @paradoxxxzero can we move this forward? This should also be ported to v13 and v15.

can I help in any way?

**Comment by dnplkndll** [comment]
using the traces_sample_rate can context be added to the transaction to reproduce the slow responses?

for instance a use case is if you have known long running calls, like to a shipping api, would like to filter the results to call not made with that certain action. Error traces have all the data. the performance samples do not. 

POST /web/dataset/call_button

image

would this be a good place to have an option like inherit_id="web.assets_backend"
```
                <script
                    src="https://browser.sentry-cdn.com/6.16.1/bundle.min.js"
                    integrity="sha384-WkFzsrcXKeJ3KlWNXojDiim8rplIj1RPsCbuv7dsLECoXY8C6Cx158CMgl+O+QKW"
                    crossorigin="anonymous"
                ></script>
                <script
                    src="https://browser.sentry-cdn.com/6.16.1/bundle.tracing.min.js"
                    integrity="sha384-hySah00SvKME+98UjlzyfP852AXjPPTh2vgJu26gFcwTlZ02/zm82SINaKTKwIX2"
                    crossorigin="anonymous"
                ></script>
                <script>
                Sentry.init({
                    dsn: "<t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.sentry_dsn', '14.0')",
                    release: "<t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.version', '14.0')"/>",
                    integrations: [new Sentry.Integrations.BrowserTracing()],
                    tracesSampleRate: <t t-esc="request.env['ir.config_parameter'].sudo().get_param('ken_erp.sentry_sample_Rate', '0.2')"/>,
                    sendClientReports: false,
                });
                </script>
```

**Comment by thapakazi** [comment]
+1, what needs to be done on this to get it merged ?

**Comment by dreispt** [comment]
We need a green TravisCI build. That's blocking this right now.

**Comment by hparfr** [comment]
Can we merge ?

**Comment by dnplkndll** [comment]
@paradoxxxzero do you have any idea how to get the header set? there are a number of integration examples now. But not clear to me the best approach here. 

https://develop.sentry.dev/sdk/performance/#header-sentry-trace

**Comment by hparfr** [comment]
@schiggi @dnplkndll @thapakazi 

Can you please mark your review status on this PR ?

**Review by dnplkndll — APPROVED** [review]

**Pull request merged**